inc/class-wpseo-admin-bar-menu.php

```diff
@@ -464,17 +464,21 @@ protected function get_singular_post() {
 		return $post;
 	}
 
-	/**
-	 * Gets the focus keyword for a given post.
-	 *
-	 * @param WP_Post $post Post object to get its focus keyword.
-	 *
-	 * @return string Focus keyword, or empty string if none available.
-	 */
-	protected function get_post_focus_keyword( WP_Post $post ) {
-		if ( apply_filters( 'wpseo_use_page_analysis', true ) !== true ) {
-			return '';
-		}
+    /**
+     * Gets the focus keyword for a given post.
+     *
+     * @param $post Post object to get its focus keyword.
+     *
+     * @return string Focus keyword, or empty string if none available.
+     */
+    protected function get_post_focus_keyword( $post ) {
+        if ( ! is_object( $post ) || ! property_exists( $post, 'ID' ) ) {
+            return '';
+        }
+
+        if ( apply_filters( 'wpseo_use_page_analysis', true ) !== true ) {
```

Comment:
Can you add this on the line before:
```

		/**
		 * Filter: 'wpseo_use_page_analysis' - Determines if the analysis should be enabled.
		 *
		 * @api        bool - Determines if the analysis should be enabled.
		 */
```

Reply:
How I read it this function returns an empty string if the page analysis is not enabled. If it is enabled it continues and returns the focus keyword. Is this correct? In that case the comments should be updated.

Reply:
This docblock is meant for the filter and that one return a boolean.